libflux/src/core/semantic/types.rs

```diff
@@ -153,16 +155,20 @@ pub enum Error {
 impl fmt::Display for Error {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match self {
-            Error::CannotUnify(t, with) => write!(f, "{} != {}", t, with),
-            Error::CannotConstrain(t, with) => write!(f, "{} is not {}", t, with),
+            Error::CannotUnify(expected, found) => write!(f, "{} != {}", found, expected),
+            Error::CannotConstrain(found, expected) => write!(f, "{} is not {}", found, expected),
```

Comment:
Should we do a similar thing here? Saying `type is not type` is the same as `type != type`. Both communicate a factual statement but do not help the user fix what is broken. 

What is the difference between CannotUnify and CannotConstrain? Is there extra context there that is helpful to the user that we should put in the error text?

Reply:
I think this one can stay as it is. This error is returned when a particular kind is expected like `Addable` but a type that is not `Addable` is found instead.

Reply:
That makes sense, I found this example from a test case:

```
type error @2:14-2:17: string is not Negatable
```

That does read well, thanks for clarifying.